### DISCUSSION TITLE
How to use serializeJson with client?

### DISCUSSION THREAD

**Discussion by piranha771** [ssilverman/QNEthernet#73] [General]
I want to use ArduinoJson's `serializeJson(doc, client);` 

Unfortunately QNEthenet's client is not recognized as a valid output stream despite it basically implements `Print`.

Is there a way to call this? I don't want to make a C-string buffer just because I don't know how to cast it, to make this work.

Additional Info: https://arduinojson.org/v7/tutorial/serialization/ Page 110

**Comment by ssilverman** [comment]
Since EthernetClient extends from Print, it should work. Can you come up with a small example and also show me the compile error you’re seeing? (I don’t know what you mean by “not recognized.”)

**Reply by piranha771** [reply]
Solved.

I'm using the `EthernetWebServer` library.  `server.client()` returns the QNEthernet client:

> inline qindesign::network::EthernetClient EthernetWebServer::client()


I was trying to use
```
    JsonDocument doc;
    doc["hello"] = "World";
    serializeJson(doc, server.client());
```

And got this error

> no instance of overloaded function "serializeJson" matches the argument listC/C++(304)
WebServer.cpp(219, 5): argument types are: (ArduinoJson::V704J1::JsonDocument, qindesign::network::EthernetClient)

For some reason I had to write it this way:
```
    JsonDocument doc;
    doc["hello"] = "World";
    EthernetClient cl = server.client();
    serializeJson(doc, cl);
```

Is it due to the inline? This can be closed nonetheless. Thanks for this great lib!

**Reply by ssilverman** [reply]
I’m curious, what happens if you do this?

```c++
serializeJson(doc, static_cast<Print&>(server.client()));
```
